lib/school/teachers.ex

```diff
@@ -31,6 +31,7 @@ defmodule School.Teachers do
   def get_teacher(id) do
     Teacher
     |> Repo.get(id)
+    |> Repo.preload(:courses)
```

Comment:
don't preload courser with get teacher, it might affect the performance of other apis that needs to get teacher without courses.
You can add `get_teacher(id, preloads \\ [])` for default behaviour